src/Microsoft.Identity.Web/AzureADB2COpenIDConnectEventHandlers.cs

```diff
@@ -41,7 +41,14 @@ public Task OnRedirectToIdentityProvider(RedirectContext context)
                 }
                 else
                 {
-                    context.ProtocolMessage.ResponseType = OpenIdConnectResponseType.CodeIdToken;
+                    if (Options.IsB2C)
+                    {
+                        context.ProtocolMessage.ResponseType = OpenIdConnectResponseType.CodeIdToken;
+                    }
+                    else
+                    {
+                        context.ProtocolMessage.ResponseType = OpenIdConnectResponseType.Code;
```

Comment:
With AAD this can work without the IDToken.
For B2C when we ask for another policy we need to have the IDToken

Reply:
i'll try to test this out today.